src/ipc/Makefile.am

```diff
@@ -71,3 +71,6 @@ DEFS += -DDEFAULT_STATEDIR=\"$(localstatedir)/run/squid\"
 
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(localstatedir)/run/squid;
+
+libipc_la_LIBADD = $(ATOMICLIB)
```

Comment:
Do you have any explanation why the library has to be added here? it should be added at the binary link stage for all Squid binaries already.

Reply:
Unfortunately, libatomic is missing when building test binaries, such as testRock and testUfs, as you can see from the failed build [output|https://build.squid-cache.org/job/5-pr-test/1959/COMPILER=clang,OS=debian-stable,label=docker-build-host/console]. I assume that It linked before the PR changes because many atomic calls were inlined. Now libipc introduces a new 'is_lock_free()' call which sometimes (on some platforms) is not inlined - so we have to add this dependency.

Reply:
That means those unit tests need $(ATOMICLIB) added to their *_LDADD list for the new code.

Reply:
Do you think that adjusting  *_LDADD  for each of testRock, testStore, testDiskIO, test_http_range, testHttpReply, testHttpRequest, testCacheMangager, testEvent, testEventLoop tests (depending on libipc) is better than fixing just libipc? Also, the implemented approach seems more reliable in case of adding new modules depending on libipc.

Reply:
Yes. This should be linked to any of the unit tests that won't build without it.

The atomic types are sometimes used in code outside of libipc and this is far from the first time this error has come up. The reason $(ATOMICLIB) exists at all is from earlier cycles spent on related build errors. That tests you find are just ones that did not need it earlier, now they do.

Reply:
AFAICT, a miscommunication resulted in what can be interpreted as an absurd conclusion:

> Eduard: Do you think that adjusting *_LDADD for each of testRock, testStore, testDiskIO, test_http_range, testHttpReply, testHttpRequest, testCacheMangager, testEvent, testEventLoop tests (depending on libipc) is better than fixing just libipc?

> Amos: Yes. This should be linked to any of the unit tests that won't build without it.

I will rephrase Amos' answer so that it makes sense for me in the context of Eduard's question and add my understanding of the current PR status:

If testX fails to build because of modifications in this PR, then this PR must fix testX build. This is true for every X.

As far as `ATOMICLIB`-related failures are concerned, fixing libipc dependencies (commit 0424152) is the best solution we now about (assuming it works). `..._la_LIBADD` exists specifically to solve these kind of problems. If it works, we should use it. Manually propagating those libipc dependencies to each and every libipc user is clearly an inferior solution in that case.

> Amos: The atomic types are sometimes used in code outside of libipc

I am not aware of any code outside libipc that uses STL atomics without being dependent on libipc. If such code exists, it is probably unrelated to this PR. We can discuss it if you insist, but that discussion is probably outside this PR scope.

> Amos: it should be added at the binary link stage for all Squid binaries already.

AFAICT, the official code adds `ATOMICLIB` to a single binary -- `squid`. @eduard-bagdasaryan, we should now remove that explicit dependency from the `squid` target, right?

We could, of course, link _all_ binaries with libatomic "just in case" but that feels like an overreaction/pollution. It is better to fix known problems using proper dependencies if we can. If I understood Eduard correctly, commit 0424152 does that. We just need to remove the explicit dependency from the `squid` binary now.

Reply:
>  we should now remove that explicit dependency 

I think we should.

Reply:
Please remove it. AFAICT, the same dependency will still be added to the same `squid` binary because the binary depends on our libipc. Thus, this removal will not have any affect on the final set of libraries the `squid` executable is linked with. Hopefully, the order of those libraries will also be OK (or does not matter).